x-pack/plugins/security_solution/public/common/store/data_table/middleware_local_storage.ts

```diff
@@ -50,21 +47,19 @@ const tableActionTypes = new Set([
   toggleDetailPanel.type,
 ]);
 
-export const createDataTableLocalStorageEpic =
-  <State>(): Epic<Action, Action, State, TimelineEpicDependencies<State>> =>
-  (action$, state$, { tableByIdSelector, storage }) => {
-    const table$ = state$.pipe(map(tableByIdSelector), filter(isNotNull));
-    return action$.pipe(
-      delay(500),
-      withLatestFrom(table$),
-      tap(([action, tableById]) => {
-        if (tableActionTypes.has(action.type)) {
-          if (storage) {
-            const tableId: TableIdLiteral = get('payload.id', action);
-            addTableInStorage(storage, tableId, tableById[tableId]);
-          }
-        }
-      }),
-      ignoreElements()
-    );
+export const dataTableLocalStorageMiddleware: (storage: Storage) => Middleware<{}, State> =
+  (storage: Storage) => (store) => (next) => (action: Action) => {
+    // perform the action
+    const ret = next(action);
+
+    // persist the data table state when a table action has been performed
+    if (tableActionTypes.has(action.type)) {
+      const tableById = dataTableSelectors.tableByIdSelector(store.getState());
+      const tableId: TableIdLiteral = get('payload.id', action);
+      if (tableById && tableById[tableId] && storage) {
+        addTableInStorage(storage, tableId, tableById[tableId]);
+      }
+    }
```

Comment:
There's a difference in the behavior, the new version stores the table on **every** action call, while the previous version had a 0.5s delay before persisting, this may have an impact if we receive multiple table actions in a short period of time, considering we are storing the entire tables redux-state, and localStorage operations are DOM blocking.
I would consider debouncing the call to `addTableInStorage` with [lodash debounce](https://lodash.com/docs/4.17.15#debounce), mirroring the previous implementation. Just to be sure we are not introducing any undesired side effects.

Reply:
sorry I did not see [this comment](https://github.com/elastic/kibana/pull/175450#discussion_r1466049421), isn't the `withLatestFrom(table$),` not taking only the last action within the 500ms delay?

Reply:
> they would result in 4 separate calls to addTableInStorage each delayed by 500ms.

Why would we want to do that? 🤔 sounds like a wrong implementation of a debouncer. Anyway, if we were not doing a proper debounce before, we are good 👍

Reply:
@semd 

I wanted to make sure I didn't make a mistake, so I created two recordings that log when `addTableInStorage` and the action filter is called in both solutions. The epic version, also calls `addTableInStorage` three times. Everything's just 500ms delayed. 

https://github.com/elastic/kibana/assets/68591/0e052c12-d400-4438-abfd-677b663dfdfa

https://github.com/elastic/kibana/assets/68591/127e916f-4d5f-4e83-ab96-5ae026546370

Reply:
That's so weird and confusing 😂. It's good we are replacing this old code. Thanks for the confirmation.